### PR TITLE
Update the request schema

### PR DESCRIPTION
**Pull request by tnagorra**

### CODE REVIEW

File: mapswipe-dashboard/app/components/MemberItem/index.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ import styles from './styles.css';
 interface Props {
     member: {
         totalMappingProjects: number;
-        totalSwipeTime: number;
+        totalSwipeTime: unknown;
```
Comment (tnagorra) [inline mapswipe-dashboard/app/components/MemberItem/index.tsx:14]: Why is this unknown?

File: mapswipe-dashboard/app/Base/utils/temporal.ts [hunk 2]
```diff
@@ -158,15 +158,14 @@ export function formatTimeDurationForSecs(
 
 // FIXME: this is a hack and needs to be fixed on the server
 export function formatTimeDuration(
-    minutes: number,
+    seconds: number,
```
[the remark this answers is not in the document]
Comment (samshara) [reply mapswipe-dashboard/app/Base/utils/temporal.ts:161]: this was changed on the server

File: mapswipe-dashboard/app/components/MemberItem/index.tsx (hunk 1 above)
Comment (samshara) [reply mapswipe-dashboard/app/components/MemberItem/index.tsx:14]: the type from server is `Scalar`

File: mapswipe-dashboard/app/views/Dashboard/index.tsx [hunk 3]
```diff
@@ -243,15 +245,13 @@ function Dashboard(props: Props) {
                     handleDateRangeChange={handleDateRangeChange}
                     className={styles.statsBoard}
                     // eslint-disable-next-line max-len
-                    contributionTimeStats={filteredCommunityStats?.filteredStats?.contributorTimeStats}
+                    contributionTimeStats={filteredCommunityStats?.filteredStats?.contributorTimeStats as ActualContributorTimeStatType[] | undefined}
```
[the remark this answers is not in the document]
Comment (samshara) [reply mapswipe-dashboard/app/views/Dashboard/index.tsx:248]: same as above

[the remark this answers is not in the document]
Comment (samshara) [reply mapswipe-dashboard/app/views/Dashboard/index.tsx:248]: Here `totalSwipeTime` is of type Scalars['TimeInSeconds'] which is of type `unknown`. That's why we have used type assertion here.

File: mapswipe-dashboard/app/views/UserDashboard/index.tsx [hunk 4]
```diff
@@ -130,26 +131,26 @@ function UserDashboard(props: Props) {
...
         [filteredUserStats],
     );
 
     const totalSwipe = userStats?.userStats?.stats?.totalSwipes;
     const totalSwipeLastMonth = userStats?.userStats?.statsLatest?.totalSwipes;
 
-    const totalSwipeTime = userStats?.userStats?.stats?.totalSwipeTime;
-    const totalSwipeTimeLastMonth = userStats?.userStats?.statsLatest?.totalSwipeTime;
+    const totalSwipeTime = userStats?.userStats?.stats?.totalSwipeTime as number;
+    const totalSwipeTimeLastMonth = userStats?.userStats?.statsLatest?.totalSwipeTime as number;
```
[the remark this answers is not in the document]
Comment (samshara) [reply mapswipe-dashboard/app/views/UserDashboard/index.tsx:153]: same as above

File: mapswipe-dashboard/app/views/UserDashboard/index.tsx [hunk 5]
```diff
@@ -262,20 +263,22 @@ function UserDashboard(props: Props) {
             <div className={styles.content}>
                 <div className={styles.container}>
                     <CalendarHeatMapContainer
-                        data={contributionData}
+                        data={contributionData as Data[]}
```
[the remark this answers is not in the document]
Comment (samshara) [reply mapswipe-dashboard/app/views/UserDashboard/index.tsx:266]: same as above

File: mapswipe-dashboard/app/views/UserDashboard/index.tsx [hunk 6]
```diff
@@ -262,20 +263,22 @@ function UserDashboard(props: Props) {
...
-                        contributionTimeStats={filteredUserStats?.user.contributionTime}
-                        projectTypeStats={filteredUserStats?.user.projectStats}
-                        organizationTypeStats={filteredUserStats?.user.organizationSwipeStats}
-                        projectSwipeTypeStats={filteredUserStats?.user.projectSwipeStats}
-                        contributions={
-                            // eslint-disable-next-line max-len
-                            filteredUserStats?.user.userGeoContribution as MapContributionType[] | undefined
-                        }
+                        // eslint-disable-next-line max-len
+                        contributionTimeStats={filteredUserStats?.userStats.filteredStats.swipeTimeByDate as ActualContributorTimeStatType[]}
```
[the remark this answers is not in the document]
Comment (samshara) [reply mapswipe-dashboard/app/views/UserDashboard/index.tsx:273]: same as above

File: mapswipe-dashboard/app/views/UserGroupDashboard/index.tsx [hunk 7]
```diff
@@ -284,23 +285,23 @@ function UserGroupDashboard(props: Props) {
             <div className={styles.content}>
                 <div className={styles.container}>
                     <CalendarHeatMapContainer
-                        data={contributionData}
+                        data={contributionData as Data[]}
```
[the remark this answers is not in the document]
Comment (samshara) [reply mapswipe-dashboard/app/views/UserGroupDashboard/index.tsx:288]: same as above

File: mapswipe-dashboard/app/views/UserGroupDashboard/index.tsx [hunk 8]
```diff
@@ -284,23 +285,23 @@ function UserGroupDashboard(props: Props) {
...
+                        // eslint-disable-next-line max-len
+                        projectSwipeTypeStats={filteredUserGroupStats?.userGroupStats.filteredStats.swipeByProjectType}
                         dateRange={dateRange}
                         handleDateRangeChange={handleDateRangeChange}
-                        contributions={
-                            // eslint-disable-next-line max-len
-                            filteredUserGroupStats?.userGroup.userGroupGeoStats as MapContributionType[]
-                        }
+                        // eslint-disable-next-line max-len
+                        contributions={filteredUserGroupStats?.userGroupStats.filteredStats.contributionByGeo as MapContributionType[]}
```
[the remark this answers is not in the document]
Comment (samshara) [reply mapswipe-dashboard/app/views/UserGroupDashboard/index.tsx:302]: same as above

File: mapswipe-dashboard/app/components/MemberItem/index.tsx (hunk 1 above)
Comment (tnagorra) [reply mapswipe-dashboard/app/components/MemberItem/index.tsx:14]: We should change this on the server

File: mapswipe-dashboard/app/views/Dashboard/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (tnagorra) [reply mapswipe-dashboard/app/views/Dashboard/index.tsx:248]: We have to fix this from the server

File: mapswipe-dashboard/app/views/UserGroupDashboard/index.tsx [hunk 9]
```diff
@@ -117,55 +127,55 @@ function UserGroupDashboard(props: Props) {
...
 
-    const totalSwipe = userGroupStats?.userGroup.stats?.totalSwipe;
-    const totalSwipeLastMonth = userGroupStats?.userGroup.userGroupLatest?.totalSwipes;
+    const totalSwipe = userGroupStats?.userGroupStats.stats.totalSwipes;
+    const totalSwipeLastMonth = userGroupStats?.userGroupStats.statsLatest.totalSwipes;
 
-    const totalSwipeTime = userGroupStats?.userGroup.stats?.totalSwipeTime;
-    const totalSwipeTimeLastMonth = userGroupStats?.userGroup.userGroupLatest?.totalSwipeTime;
+    const totalSwipeTime = userGroupStats?.userGroupStats.stats.totalSwipeTime as number;
+    const totalSwipeTimeLastMonth = userGroupStats?.userGroupStats.statsLatest.totalSwipeTime as number;
```
Comment (tnagorra) [inline mapswipe-dashboard/app/views/UserGroupDashboard/index.tsx:175]: Let's not cast this

**Pull request merged**